### ISSUE TITLE
Getting "vnull" as my version every now and then

### ISSUE THREAD

**Issue by steven-murray** [mathieudutour/github-tag-action#14]
Hi, thanks for this neat tool.

I am getting (not every time) a tag applied as "vnull". Here's my workflow:

```
name: Bump Version
on:
  push:
    branches:
      - master
jobs:
  bump-version-and-deploy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
      - name: Bump version and push tag
        uses: mathieudutour/github-tag-action@v4.2
        with:
          github_token: ${{ secrets.bump_version }}
          default_bump: false
```

Here's a snippet of the output:

```
3.0.3
v1.3.1
v2.0.0
v2.0.2
v2.0.3
v2.0.3b
v2.0.4
v2.0.5
v3.0.0
v3.0.1
v3.0.10
v3.0.11
v3.0.8
v3.0.9
vnull
/usr/bin/git rev-list --tags --topo-order --max-count=1
c7bcc9de1adf49ff03bf93a9d32e1281996a6ec4
/usr/bin/git describe --tags c7bcc9de1adf49ff03bf93a9d32e1281996a6ec4
vnull
/usr/bin/git log vnull..HEAD --pretty=format:'%s%n%b==============================================' --abbrev-commit
'Merge pull request #66 from steven-murray/fix-setup-cfg
fix: applied changes suggested by @webknjaz=============================================='::debug::Setting previous_tag to: vnull
Analyzing commit: Merge pull request #66 from steven-murray/fix-setup-cfg
fix: applied changes suggested by @webknjaz
The commit should not trigger a release
Analysis of 1 commits complete: no release
/usr/bin/git tag -l vnull
vnull
```

I can't quite put my finger on what is making this happen?

EDIT: I guess the most likely culprit is me setting `default_bump` to `false`. Is there a way to *not* bump the version if no appropriately tagged commits are found?

**Comment by mathieudutour** [comment]
fixed in v4.4

**Comment by amadeu01** [comment]
hey @mathieudutour I'm experiencing the same issue in 4.5 

```
/usr/bin/git fetch --tags
/usr/bin/git tag
build-number-3
v0.0.1
v0.0.2
v0.0.3
v0.0.4
v0.0.5
v0.1.0
vnull
/usr/bin/git rev-list --tags --topo-order --max-count=1
0a0b23629705d0146078bb9dc92d2c5352dfdc83
/usr/bin/git describe --tags 0a0b23629705d0146078bb9dc92d2c5352dfdc83
build-number-3
/usr/bin/git log build-number-3..HEAD --pretty=format:'%s%n%b|commit-hash:%h==============================================' --abbrev-commit
'fix: Fix CI
|commit-hash:e31673c=============================================='::debug::Setting previous_tag to: build-number-3
Analyzing commit: fix: Fix CI
The release type for the commit is patch
Analysis of 1 commits complete: patch release
/usr/bin/git tag -l vnull
vnull
```

I believe that it might be due the `einaregilsson/build-number@v2 ` action that is generating a tag build-number
